astropy/table/operations.py

```diff
@@ -129,6 +130,86 @@ def join(left, right, keys=None, join_type='inner',
     return out
 
 
+def setdiff(table1, table2, keys=None):
+    """
+    Take a set difference of table rows.
+
+    The row set difference will contain all rows in table1 that are not
+    present in table2.
+
+    Parameters
+    ----------
+    table1 : Table object
+        table1 is on the left side of the set difference.
+    table2 : Table object
+        table2 is on the right side of the set difference.
+    keys : str or list of str
+        Name(s) of column(s) used to match rows of left and right tables.
+        Default is to use all columns which are common to both tables.
+
+    Returns
+    -------
+    diff_table : `~astropy.table.Table` object
+        New table containing the set difference between tables. If the set
+        difference is none, an empty table will be returned.
+
+    Examples
+    --------
+    To get a set difference between two tables do::
+
+      >>> from astropy.table import setdiff, Table
+      >>> t1 = Table({'a': [1, 4, 9], 'b': ['c', 'd', 'f']}, names=('a', 'b'))
+      >>> t2 = Table({'a': [1, 5, 9], 'b': ['c', 'b', 'f']}, names=('a', 'b'))
+      >>> print(t1)
+       a   b
+      --- ---
+        1   c
+        4   d
+        9   f
+      >>> print(t2)
+       a   b
+      --- ---
+        1   c
+        5   b
+        9   f
+      >>> print(setdiff(t1, t2))
+       a   b
+      --- ---
+        4   d
+
+      >>> print(setdiff(t2, t1))
+       a   b
+      --- ---
+        5   b
+    """
+    # Make a light internal copy to avoid touching table2
+    table2 = table2.copy(copy_data=False)
+    table2['__index__'] = np.zeros(len(table2), dtype=np.uint8)  # dummy column
+    if keys is None:
+        keys = table1.colnames
+
+        # Check that all keys in table2 are in table1, if not, raise error
+        for key in table1.keys():
+            if key not in table2.keys():
+                raise ValueError("The {} column is in table1 but not in "
+                                 "table2, cannot take a set difference."
+                                 .format(key))
+
+
+    t12 = _join(table1, table2, join_type='left', keys=keys,
+                metadata_conflicts='silent')
+
+    if t12.masked:
+        diff = t12['__index__'].mask
+        t12_diff = t12[diff]
+    else:
+        t12_diff = t12[[]]
+
+    t12_diff.keep_columns(keys)
```

Comment:
I ended up deciding that it was better to do away with the masked columns. I don't see them every being that useful.  But forgot to update the doc string to correspond.